src/components/transport_manager/src/transport_adapter/transport_adapter_impl.cc

```diff
@@ -774,7 +774,7 @@ void TransportAdapterImpl::RemoveFinalizedConnection(
     const DeviceUID& device_handle, const ApplicationHandle& app_handle) {
   const DeviceUID device_uid = device_handle;
   LOG4CXX_AUTO_TRACE(logger_);
-  sync_primitives::AutoWriteLock lock(connections_lock_);
+  connections_lock_.AcquireForWriting();
```

Comment:
@Ypostolov 
1) potential dead lock  if (it_conn == connections_.end()) and if (info.state != ConnectionInfo::FINALISING) 
2) logging under critical section

Reply:
1. Used scoped lock to avoid such situations. See 002e4222f6ca70b2d66bd3dc3da3850d9f6b325b
2. Using logger inside critical section - its a common case in SDL project and that's OK.